SEP-0012.md

```diff
@@ -0,0 +1,343 @@
+
+        """
+
+    @abc.abstractmethod
+    def crop(self,
+             lower_corner: Iterable[Union[u.Quantity, float]],
+             upper_corner: Iterable[Union[u.Quantity, float]] = None,
+             units: Iterable[Union[str, u.Unit]] = None,
+             wcs: BaseHighLevelWCS = None):
+        """
+        Crops an NDCube given minimum values and interval widths along axes.
+
+        Parameters
+        ----------
+        lower_corner
+            The minimum desired values along each relevant axis after cropping
+            described in physical units consistent with the NDCube's wcs object.
+            The length of the iterable must equal the number of data dimensions
+            and must have the same order as the data.
+
+        upper_corner
+            The maximum desired values along each relevant axis after cropping
+            described in physical units consistent with the NDCube's wcs object.
+            The length of the iterable must equal the number of data dimensions
+            and must have the same order as the data.
+
+        units
+            If the inputs are set without units, the user must set the units
+            inside this argument as `str` or `~astropy.units.Unit` objects.
+            The length of the iterable must equal the number of data dimensions
+            and must have the same order as the data.
+
+        wcs
+            The WCS to use to calculate the pixel coordinates based on the input.
+            Will default to the ``.wcs`` property if not given. While any valid WCS
+            could be used it is expected that either the ``.wcs`` or
+            ``.extra_coords.wcs`` properties will be used.
+
+        Returns
+        -------
+        result: NDCube
+        """
+    
+```
+
+## Extra Coords
+
+The extra coords property of the `NDCube` class provides a core unique selling point of the package.
+It is an easy way to add extra variables which vary with one, or more, of the axes of the data.
+The [proposed implementation](https://github.com/sunpy/ndcube/pull/271) of `ExtraCoords` is heavily dependant on [gWCS](https://gwcs.readthedocs.io/en/latest/), however, the goal for this SEP is to formulate the high level API and not the implementation details.
+This will hopefully allow future iteration on the implementation without breaking the functionality of the `NDCube` class.
+
+```python
+
+class ExtraCoordsABC(abc.ABC):
+    """
+    A representation of additional world coordinates associated with pixel axes.
+
+    ExtraCoords can be initialised by either specifying a
+    `~astropy.wcs.wcsapi.LowLevelWCS` object and a ``mapping``, or it can be
+    built up by specifying one or more lookup tables.
+
+    Parameters
+    ----------
+    wcs
+        The WCS specifying the extra coordinates.
+    mapping
+       The mapping between the array dimensions and pixel dimensions in the
+       extra coords object. This is an iterable of ``(array_dimension, pixel_dimension)`` pairs
```

Comment:
What if the extra coord maps from one array dimension to more than one pixel dimension or vice versa?  Can/Should this be supported in this API?

Reply:
It can not. It can map from one array dimension to many world dimensions, but only a 1:1 mapping between array and pixel inputs is possible. Not having that restriction would completely remove the link back to the actual array in the NDCube.